Library/OcBootManagementLib/OcBootManagementLib.c

```diff
@@ -544,27 +551,39 @@ OcLaunchAppleBootPicker (
   } else if (FirmwareUI->Revision != APPLE_FIRMWARE_USER_INTERFACE_PROTOCOL_REVISION) {
     DEBUG ((
       DEBUG_INFO,
-      "OC: Launch Apple picker incompatible FirmwareUI protocol revision %u != %u\n",
+      "OC: Unlock Apple picker incompatible FirmwareUI protocol revision %u != %u\n",
       FirmwareUI->Revision,
       APPLE_FIRMWARE_USER_INTERFACE_PROTOCOL_REVISION
       ));
     Status = EFI_UNSUPPORTED;
   }
 
   if (!EFI_ERROR (Status)) {
+    UIVars = &BASE_CR (FirmwareUI, APPLE_FIRMWARE_UI_LAYOUT, UIProtocol)->UIVars;
+
     //
-    // Location of relevant byte variable within loaded driver.
+    // Also acts as partial check that the memory is not something else entirely.
     //
-    aGopAlreadyConnected = (VOID *)((UINT8 *)FirmwareUI + sizeof (APPLE_FIRMWARE_USER_INTERFACE_PROTOCOL));
-
-    if (*aGopAlreadyConnected != 1) {
-      DEBUG ((DEBUG_WARN, "OC: Cannot force reconnect Apple GOP %u\n", *aGopAlreadyConnected));
+    if (*((UINTN *)(&UIVars->mGopAlreadyConnected)) != 1) {
```

Comment:
This makes me even more convinced it is not a boolean. Could you provide some efi file?
Also, that would need ReadUnaligned...

Reply:
Think this is resolved in offline chat - will fix the packed issue - ty!